### PR TITLE
only update on json file changes

### PR DESCRIPTION
**Pull request by kruxErinWillingham**
Complete rewrite of assigning values from json
will only push updates to marathon if changes have been made

side note: note the print statement in the new block of code, there seems to be a bug in the logger that won't let me send this info to log so I'm using print instead. I'm guessing it has to do with sending multiple values. Feature update for later.

### CODE REVIEW

File: krux_marathon_api/marathonapi.py [hunk 1]
```diff
@@ -66,42 +64,41 @@ def read_config_file(self, config_file):
...
-        self.logger.debug("Marathon app object: {}".format(marathon_app_result))
+        ### value for if there is a change in our json file from server values
+        changes_in_json = False
+
+        ### iterate through our json file's values and compare them to the values
+        ### on the marathon server
+        for attribute, value in config_file_data.iteritems():
+            ### upgrade_strategy value returns a class an not a json, extra work
+            ### needs to be done to convert this to a comparable value
+            if attribute == 'upgrade_strategy':
```
Comment (thekad) [inline krux_marathon_api/marathonapi.py]: I believe other attributes are returned as MarathonResource sub classes. How about instead of possibly chasing attribute by attribute, you do something like:

```python
if isinstance(value, marathon.MarathonObject):
  marathon_app_result_original = getattr(marathon_app_result, attribute).to_json()
  value_json = json.dumps(value)
```

etc etc, makes sense?

File: krux_marathon_api/marathonapi.py [hunk 2]
```diff
@@ -66,42 +64,41 @@ def read_config_file(self, config_file):
...
+                ### the marathon api allows us to convert this attribute to json
+                marathon_app_result_original = getattr(marathon_app_result, attribute).to_json()
+                value_json = json.dumps(value)
+                if marathon_app_result_original == value_json:
+                    #print attribute, ': ', marathon_app_result_original, ' is equal to ', value_json
+                    pass
+                else:
+                    changes_in_json = True
+                    setattr(marathon_app_result, attribute, value)
+                    print 'Updating ', attribute, ' from \n', marathon_app_result_original, '\nto \n', value_json
```
Comment (thekad) [inline krux_marathon_api/marathonapi.py]: get rid of these prints and use logging.debug

Comment (kruxErinWillingham) [inline krux_marathon_api/marathonapi.py]: look at my comment in this PR, this is because of some sort of logging bug. I might be able to pass each value one at a time, but that would be super annoying.  Any other suggestions?

File: krux_marathon_api/marathonapi.py [hunk 3]
```diff
@@ -66,42 +65,42 @@ def read_config_file(self, config_file):
...
-        if config_file_data.get("port_definitions"):
-            marathon_app_result.port_definitions     = config_file_data["port_definitions"]
-            marathon_app_result.ports                = None
-        marathon_app_result.require_ports            = config_file_data["require_ports"]
-        marathon_app_result.store_urls               = config_file_data["store_urls"]
-        marathon_app_result.upgrade_strategy         = config_file_data["upgrade_strategy"]
-        marathon_app_result.uris                     = config_file_data["uris"]
-        marathon_app_result.user                     = config_file_data["user"]                 #updates confirmed
-        self.logger.debug("Marathon app object: {}".format(marathon_app_result))
+        ### value for if there is a change in our json file from server values
```
Comment (dpricha89) [inline krux_marathon_api/marathonapi.py]: Nit: Why did you remove the doc string?
